src/node.cpp

```diff
@@ -377,6 +378,14 @@ int main(int argc, char * argv[]) {
                              angle_min, angle_max, max_distance,
                              frame_id);
             }
+        } else {
+            if (!checkRPLIDARHealth(drv)) {
+                ROS_WARN_THROTTLE(1, "RPLIDAR disconnected. Attempting reconnection.");
+                RPlidarDriver::DisposeDriver(drv);
+                ros::Duration(1).sleep();
+                goto connect;
+            } 
+            ROS_WARN_THROTTLE(1, "Grab scan data failed.");
```

Comment:
not needed, I just wanted you to verify if this happens on normal run but it does not need to be committed